presto-docs/src/main/sphinx/connector/elasticsearch.rst

```diff
@@ -0,0 +1,137 @@
+
+.. note::
+
+    It is highly recommend to use Elasticsearch 5.1.2 or later.
+
+Configuration
+-------------
+
+To configure the Elasticsearch connector, create a catalog properties file
+``etc/catalog/elasticsearch.properties`` with the following contents,
+replacing the properties as appropriate:
+
+.. code-block:: none
+
+    connector.name=elasticsearch
+    elasticsearch.table-names=schema.table1,schema.table2
+    elasticsearch.default-schema=default
+    elasticsearch.table-description-dir=etc/elasticsearch/
+    elasticsearch.scroll-size=1000
+    elasticsearch.scroll-time=60000
+
+
+Multiple Elasticsearch Clusters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can have as many catalogs as you need, so if you have additional
+Elasticsearch clusters, simply add another properties file to ``etc/catalog``
+with a different name (making sure it ends in ``.properties``). For
+example, if you name the property file ``users.properties``, Presto
+will create a catalog named ``users`` using the configured connector.
+
+Configuration Properties
+------------------------
+
+The following configuration properties are available:
+
+======================================= ==============================================================================
+Property Name                           Description
+======================================= ==============================================================================
+``elasticsearch.table-names``           List of all tables provided by the catalog
+``elasticsearch.default-schema``        Default schema name for tables
+``elasticsearch.table-description-dir`` Directory containing topic description files
+``elasticsearch.scroll-size``           Maximum number of hits to be returned with each batch of Elasticsearch scroll
+``elasticsearch.scroll-time``           Number of milliseconds the Elasticsearch would keep live for a search context
+======================================= ==============================================================================
+
+``elasticsearch.table-names``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Comma-separated list of all tables provided by this catalog. A table name
+can be unqualified (simple name) and will be put into the default schema
+(see below) or qualified with a schema name (``<schema-name>.<table-name>``).
+
+For each table defined here, a table description file (see below) must exist.
+
+This property is required; there is no default and at least one table must be defined.
+
+``elasticsearch.default-schema``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Defines the schema which will contain all tables that were defined without
+a qualifying schema name.
+
+This property is optional; the default is ``default``.
+
+``elasticsearch.table-description-dir``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+References a folder within Presto deployment that holds one or more JSON
```

Comment:
`References a directory in the Presto deployment directory that contains one or more JSON files with table descriptions (must end with ``.json``).`